### PR TITLE
Fastify-2 blog post

### PR DESCRIPTION
**Pull request by tmcw**
Blog post about the upsides of our fastify transition - I'll update the Townie screenshot with, probably, a video, once a PR is merged on that side.

It's kind of a casual tone, which was just what my gut was feeling for this.

### CODE REVIEW

**Comment by stevekrouse** [comment]
Great! I'll add some comments now

I wonder why the cloudflare preview thing didn't happen on this PR

File: src/content/blog/fastify-2.mdx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+title: "Moving to fastify pt. 2: Docs, an SDK and AI tools"
+description: |
+  Taking advantage of our typed REST API to build a platform around
+  Val Town.
+pubDate: July 12, 2024
```
Comment (stevekrouse) [inline src/content/blog/fastify-2.mdx:6]: Reminder to update :)

File: src/content/blog/fastify-2.mdx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+author: Tom MacWright
+---
+
+The last time, I [wrote about how we were switching from express to fastify][last_post]
+for Val Town’s REST API. To recap, Fastify gives us a way to strictly
```
Comment (stevekrouse) [inline src/content/blog/fastify-2.mdx:11]:
```suggestion
We recently [switched from express to fastify][last_post]
for Val Town’s REST API. Fastify gives us a way to strictly
```

File: src/content/blog/fastify-2.mdx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+is basically as simple as this:
+
+```html
+<script id="api-reference" data-url="./openapi.documented.json"></script>
+<script src="https://cdn.jsdelivr.net/npm/@scalar/api-reference"></script>
+```
+
+### A TypeScript SDK
+
+Are you not entertained? How about [a new TypeScript SDK](https://github.com/val-town/sdk)?
```
Comment (stevekrouse) [inline src/content/blog/fastify-2.mdx:41]:
```suggestion
How about [a new TypeScript SDK](https://github.com/val-town/sdk)?
```

Comment (stevekrouse) [reply src/content/blog/fastify-2.mdx:41]: (I didn't find the joke funny, I guess?)

File: src/content/blog/fastify-2.mdx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+// Use it
+console.log(await valTown.me.profile.retrieve());
+```
+
+The SDK covers all of our major API methods, and has robust TypeScript
+types for everything. Methods in our API that are paginated, like listing
+comments, vals, or search results, are implemented as async iterators
+in the SDK so you don't even have to implement any pagination logic.
+
+Just install it.
```
Comment (stevekrouse) [inline src/content/blog/fastify-2.mdx:57]: I think we should prioritize showing off its use in Val Town first and foremost (link to a val or two?) and then say of course it works in all the other js runtimes

File: src/content/blog/fastify-2.mdx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+sentience, fortunately/unfortunately, but it can already write some pretty
+decent code.
+
+But it couldn't really interact with Val Town until now. Townie just got
+the ability to call our API directly, which means that it can do things like
+creating vals, searching for existing code, updating readmes – anything
+that the API can do, Townie can do too.
+To be safe, Townie will run `GET` requests against the API freely, but asks
+for permission before running `POST`, `PUT`, or `DELETE` requests.
+
```
Comment (stevekrouse) [inline src/content/blog/fastify-2.mdx:84]: A video here could be awesome, but not necessary

File: src/content/blog/fastify-2.mdx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+Townie knows the whole layout of the API because OpenAPI 3.1 is
+[compatible with JSON Schema][oa31], and
+the [OpenAI Function calling][calling]
+system is based on JSON Schema-defined parameters. So we can fairly smoothly
+translate from our OpenAPI specification for our REST API, into a definition
+of functions that OpenAI can call – along with LLM-friendly descriptions
+for every function, parameter, and return value – and voila, the robot
+can use the machine. Avoiding OpenAPI-OpenAI mixup typos was one of the more
+mentally challenging parts of implementing this.
+
```
Comment (stevekrouse) [inline src/content/blog/fastify-2.mdx:94]: Feels a bit weird to end on that note. Some ideas:

* any final thoughts on Fastify/OpenAPI/OpenAI, migrations, etc
* Tips for others
* Future directions

File: src/content/blog/fastify-2.mdx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+for permission before running `POST`, `PUT`, or `DELETE` requests.
+
+Townie knows the whole layout of the API because OpenAPI 3.1 is
+[compatible with JSON Schema][oa31], and
+the [OpenAI Function calling][calling]
+system is based on JSON Schema-defined parameters. So we can fairly smoothly
+translate from our OpenAPI specification for our REST API, into a definition
+of functions that OpenAI can call – along with LLM-friendly descriptions
+for every function, parameter, and return value – and voila, the robot
+can use the machine. Avoiding OpenAPI-OpenAI mixup typos was one of the more
```
Comment (jxnblk) [inline src/content/blog/fastify-2.mdx:93]: > and voila, the robot can use the machine.

I really like this line

Comment (tmcw) [reply src/content/blog/fastify-2.mdx]: Thx!

**Pull request merged**